From c656d523de03ab1816dae08943d205b9a4126432 Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 7 Jan 2016 17:08:35 +0100
Subject: [PATCH 10/11] S Core: Add init of second SD interface

The SD2 interface will be muxed and initialized. There are although some
outlines added to give the user a hint weather the U-Boot was loaded
from SPI or from SD-Card

Ticket #871
---
 board/freescale/mx6sxscore/mx6sxscore.c | 35 ++++++++++++++++++++++++++-------
 1 file changed, 28 insertions(+), 7 deletions(-)

diff --git a/board/freescale/mx6sxscore/mx6sxscore.c b/board/freescale/mx6sxscore/mx6sxscore.c
index a0fd398..490955b 100644
--- a/board/freescale/mx6sxscore/mx6sxscore.c
+++ b/board/freescale/mx6sxscore/mx6sxscore.c
@@ -1,9 +1,15 @@
 /*
  * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ * Copyright (C) 2015, 2016 sitec systems GmbH
  *
  * SPDX-License-Identifier:	GPL-2.0+
  */
 
+// This is the board file for the SCore SoM and the SCore evaluation board.
+// There are two possiblities for booting the board:
+//   - from an prepared SD-Card on SD3
+//   - from the on module SPI NOR flash
+
 #include <asm/arch/clock.h>
 #include <asm/arch/iomux.h>
 #include <asm/arch/imx-regs.h>
@@ -157,7 +163,17 @@ static iomux_v3_cfg_t const usdhc3_pads[] = {
     MX6SX_PAD_SD3_DATA5__GPIO7_IO_7 | MUX_PAD_CTRL(NO_PAD_CTRL),
 };
 
+static iomux_v3_cfg_t const usdhc2_pads[] = {
+	MX6SX_PAD_SD2_CLK__USDHC2_CLK | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6SX_PAD_SD2_CMD__USDHC2_CMD | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6SX_PAD_SD2_DATA0__USDHC2_DATA0 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6SX_PAD_SD2_DATA1__USDHC2_DATA1 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6SX_PAD_SD2_DATA2__USDHC2_DATA2 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+	MX6SX_PAD_SD2_DATA3__USDHC2_DATA3 | MUX_PAD_CTRL(USDHC_PAD_CTRL),
+};
+
 static struct fsl_esdhc_cfg usdhc_cfg[2] = {
+    {USDHC2_BASE_ADDR, 0, 4},
     {USDHC3_BASE_ADDR, 0, 4},
 };
 
@@ -218,15 +234,21 @@ int board_mmc_init(bd_t *bis)
     /*
      * According to the board_mmc_init() the following map is done:
      * (U-boot device node)    (Physical Port)
-     * mmc0                    USDHC3
+     * mmc0                    USDHC2
+     * mmc1                    USDHC3
      */
     for (i = 0; i < CONFIG_SYS_FSL_USDHC_NUM; i++) {
         switch (i) {
             case 0:
                 imx_iomux_v3_setup_multiple_pads(
+                        usdhc3_pads, ARRAY_SIZE(usdhc2_pads));
+                usdhc_cfg[0].sdhc_clk = mxc_get_clock(MXC_ESDHC2_CLK);
+                break;
+            case 1:
+                imx_iomux_v3_setup_multiple_pads(
                         usdhc3_pads, ARRAY_SIZE(usdhc3_pads));
                 gpio_direction_input(USDHC3_CD_GPIO);
-                usdhc_cfg[0].sdhc_clk = mxc_get_clock(MXC_ESDHC3_CLK);
+                usdhc_cfg[1].sdhc_clk = mxc_get_clock(MXC_ESDHC3_CLK);
                 break;
             default:
                 printf("Warning: you configured more USDHC controllers"
@@ -531,11 +553,13 @@ int board_init(void)
 
 #ifdef CONFIG_SYS_USE_SPINOR
 	setup_spinor();
+    printf("Hello World from SPINOR\n");
+#else
+    printf("Hello World from SD Card\n");
 #endif
 
     setup_gpmi_nand();
     setup_gpio();
-    printf("Hello World from SPINOR\n");
     return 0;
 }
 
@@ -561,10 +585,7 @@ int board_late_init(void)
     if (ret)
         return -1;
 #endif
-
-#ifdef CONFIG_ENV_IS_IN_MMC
-    board_late_mmc_init();
-#endif
+    /*board_late_mmc_init();*/
 
     return 0;
 }
-- 
1.9.1

